tracer/src/Datadog.Trace/Configuration/TracerSettings.cs

```diff
@@ -58,26 +59,45 @@ public TracerSettings(IConfigurationSource? source)
             var commaSeparator = new[] { ',' };
             source ??= NullConfigurationSource.Instance;
 
-            Environment = source.GetString(ConfigurationKeys.Environment);
-
-            ServiceName = source.GetString(ConfigurationKeys.ServiceName) ??
-                          // backwards compatibility for names used in the past
-                          source.GetString("DD_SERVICE_NAME");
-
-            ServiceVersion = source.GetString(ConfigurationKeys.ServiceVersion);
-
-            GitCommitSha = source.GetString(ConfigurationKeys.GitCommitSha);
-
-            GitRepositoryUrl = source.GetString(ConfigurationKeys.GitRepositoryUrl);
-
-            GitMetadataEnabled = source.GetBool(ConfigurationKeys.GitMetadataEnabled) ?? true;
-
-            TraceEnabled = source.GetBool(ConfigurationKeys.TraceEnabled) ??
-                           // default value
-                           true;
-
-            var disabledIntegrationNames = source.GetString(ConfigurationKeys.DisabledIntegrations)
-                                                 ?.Split(new[] { ';' }, StringSplitOptions.RemoveEmptyEntries) ??
+            Environment = new ConfigurationBuilder(source, Telemetry)
+                         .WithKeys(ConfigurationKeys.Environment)
+                         .AsString()
+                         .Get();
+
+            // backwards compatibility for names used in the past
```

Comment:
This comment is for the `"DD_SERVICE_NAME"` below, so now it's misplaced. I think we can just remove it, tbh.